drivers/oled/oled_driver.c

```diff
@@ -152,6 +154,10 @@ static void InvertCharacter(uint8_t *cursor) {
 }
 
 bool oled_init(uint8_t rotation) {
+#ifdef USE_I2C
```

Comment:
```suggestion
#if defined(USE_I2C) && defined(SPLIT_KEYBOARD)
```

Also, this should probably go into oled_task, (line 687~)

Reply:
I  am not sure. It's up to you. If you wish, I can add changes to oled_task.
I agree that adding check in oled_task gives more confidence that code will not execute on the slave half.

I decided not to change oled_task because the logic in oled_task is based on oled_initialized variable.  
Oled_initialized set true only at the end of oled_init
https://github.com/qmk/qmk_firmware/blob/ce47d1907a02f857f764ddc396053dd4c56a837f/drivers/oled/oled_driver.c#L220
So if we return early from oled_init, then  oled_initialized is always false.
And oled_task will always return early
https://github.com/qmk/qmk_firmware/blob/ce47d1907a02f857f764ddc396053dd4c56a837f/drivers/oled/oled_driver.c#L688